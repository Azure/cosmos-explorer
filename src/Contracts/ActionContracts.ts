/**
 * Data Explorer tab kinds
 */
export enum TabKind {
  SQLDocuments,
  MongoDocuments,
  MongoSchema,
  TableEntities,
  Graph,
  SQLQuery,
  ScaleSettings,
}

/**
 * Data Explorer pane kinds
 */
export enum PaneKind {
  AddCollection,
  CassandraAddCollection,
  DeleteCollection,
  DeleteDatabase,
  GlobalSettings,
  AdHocAccess,
  SwitchDirectory,
}

/**
 * Parameters to pass to DataExplorer in order to have it perform a given action
 */
export interface DataExplorerAction {
  actionType: ActionType | string;
}

/**
 * Open tab action
 */
export interface OpenTab extends DataExplorerAction {
  tabKind: TabKind | string;
}

/**
 * Open collection tab action
 */
export interface OpenCollectionTab extends OpenTab {
  databaseResourceId: string;
  collectionResourceId: string;
}

/**
 * Open query tab action
 */
export interface OpenQueryTab extends OpenCollectionTab {
  query: QueryInfo;
}

/**
 * Query info will be looked at in the order given in this interface (i.e. if text is provided, everything else is ignored)
 */
export interface QueryInfo {
  text?: string;
  partitionKeys?: string[];
}

/**
 * Open pane action
 */
export interface OpenPane extends DataExplorerAction {
  paneKind: PaneKind | string;
}

export interface OpenSampleNotebook extends DataExplorerAction {
  path: string;
}

/**
 * The types of actions that the DataExplorer supports performing upon opening.
 */
export enum ActionType {
  OpenTab,
  OpenCollectionTab,
  OpenPane,
  TransmitCachedData,
  OpenSampleNotebook,
}
