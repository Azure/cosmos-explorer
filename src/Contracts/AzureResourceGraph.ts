export interface QueryRequestOptions {
  $skipToken?: string;
  $top?: number;
  allowPartialScopes: boolean;
  subscriptions?: string[];
}

export interface QueryResponse {
  $skipToken: string;
  count: number;
  data: any;
  resultTruncated: boolean;
  totalRecords: number;
}
