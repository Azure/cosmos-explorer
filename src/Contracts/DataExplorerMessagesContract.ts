import { FabricMessageTypes } from "./FabricMessageTypes";

// This is the current version of these messages
export const DATA_EXPLORER_RPC_VERSION = "3";

// Data Explorer to Fabric
export type DataExploreMessageV3 =
  | {
      type: FabricMessageTypes.Ready;
      id: string;
      params: [string]; // version
    }
  | {
      type: FabricMessageTypes.GetAuthorizationToken;
      id: string;
      params: GetCosmosTokenMessageOptions[];
    }
  | {
      type: FabricMessageTypes.GetAllResourceTokens;
      id: string;
    }
  | {
      type: FabricMessageTypes.OpenSettings;
      settingsId: string;
    };
export interface GetCosmosTokenMessageOptions {
  verb: "connect" | "delete" | "get" | "head" | "options" | "patch" | "post" | "put" | "trace";
  resourceType: "" | "dbs" | "colls" | "docs" | "sprocs" | "pkranges";
  resourceId: string;
}
