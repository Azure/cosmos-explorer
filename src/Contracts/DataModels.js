"use strict";
exports.__esModule = true;
exports.SparkClusterEndpointKind = exports.ConflictResolutionMode = exports.ApiKind = void 0;
var ApiKind;
(function (ApiKind) {
    ApiKind[ApiKind["SQL"] = 0] = "SQL";
    ApiKind[ApiKind["MongoDB"] = 1] = "MongoDB";
    ApiKind[ApiKind["Table"] = 2] = "Table";
    ApiKind[ApiKind["Cassandra"] = 3] = "Cassandra";
    ApiKind[ApiKind["Graph"] = 4] = "Graph";
    ApiKind[ApiKind["MongoDBCompute"] = 5] = "MongoDBCompute";
})(ApiKind = exports.ApiKind || (exports.ApiKind = {}));
var ConflictResolutionMode;
(function (ConflictResolutionMode) {
    ConflictResolutionMode["Custom"] = "Custom";
    ConflictResolutionMode["LastWriterWins"] = "LastWriterWins";
})(ConflictResolutionMode = exports.ConflictResolutionMode || (exports.ConflictResolutionMode = {}));
var SparkClusterEndpointKind;
(function (SparkClusterEndpointKind) {
    SparkClusterEndpointKind["SparkUI"] = "SparkUI";
    SparkClusterEndpointKind["HistoryServerUI"] = "HistoryServerUI";
    SparkClusterEndpointKind["Livy"] = "Livy";
})(SparkClusterEndpointKind = exports.SparkClusterEndpointKind || (exports.SparkClusterEndpointKind = {}));
