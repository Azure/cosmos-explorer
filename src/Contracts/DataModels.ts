import { CapacityMode, ConnectionStatusType, ContainerStatusType } from "../Common/Constants";

export interface ArmEntity {
  id: string;
  name: string;
  location: string;
  type: string;
  kind: string;
  tags?: Tags;
}

export interface DatabaseAccount extends ArmEntity {
  properties: DatabaseAccountExtendedProperties;
  systemData?: DatabaseAccountSystemData;
}

export interface DatabaseAccountSystemData {
  createdAt: string;
}

export interface DatabaseAccountExtendedProperties {
  documentEndpoint?: string;
  disableLocalAuth?: boolean;
  tableEndpoint?: string;
  gremlinEndpoint?: string;
  cassandraEndpoint?: string;
  configurationOverrides?: ConfigurationOverrides;
  capabilities?: Capability[];
  enableMultipleWriteLocations?: boolean;
  mongoEndpoint?: string;
  readLocations?: DatabaseAccountResponseLocation[];
  writeLocations?: DatabaseAccountResponseLocation[];
  enableFreeTier?: boolean;
  enableAnalyticalStorage?: boolean;
  enableMaterializedViews?: boolean;
  isVirtualNetworkFilterEnabled?: boolean;
  ipRules?: IpRule[];
  privateEndpointConnections?: unknown[];
  capacity?: { totalThroughputLimit: number };
  capacityMode?: CapacityMode;
  locations?: DatabaseAccountResponseLocation[];
  postgresqlEndpoint?: string;
  publicNetworkAccess?: string;
  enablePriorityBasedExecution?: boolean;
  vcoreMongoEndpoint?: string;
}

export interface DatabaseAccountResponseLocation {
  documentEndpoint: string;
  failoverPriority: number;
  id: string;
  locationId: string;
  locationName: string;
  provisioningState: string;
}

export interface IpRule {
  ipAddressOrRange: string;
}

export interface ConfigurationOverrides {
  EnableBsonSchema: string;
}

export interface Capability {
  name: string;
  description: string;
}

export interface AccessInputMetadata {
  accountName: string;
  apiEndpoint: string;
  apiKind: number;
  documentEndpoint: string;
  expiryTimestamp: string;
  mongoEndpoint?: string;
}

export enum ApiKind {
  SQL = 0,
  MongoDB,
  Table,
  Cassandra,
  Graph,
  MongoDBCompute,
}

export interface GenerateTokenResponse {
  readWrite: string;
  read: string;
}

export interface Subscription {
  uniqueDisplayName?: string;
  displayName: string;
  subscriptionId: string;
  tenantId?: string;
  state: string;
  subscriptionPolicies?: SubscriptionPolicies;
  authorizationSource?: string;
}

export interface SubscriptionPolicies {
  locationPlacementId: string;
  quotaId: string;
  spendingLimit?: string;
}

export interface Resource {
  _rid: string;
  _self: string;
  _etag: string;
  _ts: number | string;
  id: string;
}

export interface IType {
  name: string;
  code: number;
}

export interface IDataField {
  dataType: IType;
  hasNulls: boolean;
  isArray: boolean;
  schemaType: IType;
  name: string;
  path: string;
  maxRepetitionLevel: number;
  maxDefinitionLevel: number;
}

export interface ISchema {
  id: string;
  accountName: string;
  resource: string;
  fields: IDataField[];
}

export interface ISchemaRequest {
  id: string;
  subscriptionId: string;
  resourceGroup: string;
  accountName: string;
  resource: string;
  status: string;
}

export interface Collection extends Resource {
  // Only in Mongo collections loaded via ARM
  shardKey?: {
    [key: string]: string;
  };
  defaultTtl?: number;
  indexingPolicy?: IndexingPolicy;
  partitionKey?: PartitionKey;
  statistics?: Statistic[];
  uniqueKeyPolicy?: UniqueKeyPolicy;
  conflictResolutionPolicy?: ConflictResolutionPolicy;
  changeFeedPolicy?: ChangeFeedPolicy;
  analyticalStorageTtl?: number;
  geospatialConfig?: GeospatialConfig;
  vectorEmbeddingPolicy?: VectorEmbeddingPolicy;
  fullTextPolicy?: FullTextPolicy;
  schema?: ISchema;
  requestSchema?: () => void;
  computedProperties?: ComputedProperties;
  materializedViews?: MaterializedView[];
  materializedViewDefinition?: MaterializedViewDefinition;
}

export interface CollectionsWithPagination {
  continuationToken?: string;
  collections?: Collection[];
}

export interface Database extends Resource {
  collections?: Collection[];
}

export interface DocumentId extends Resource {}

export interface ConflictId extends Resource {
  resourceId?: string;
  resourceType?: string;
  operationType?: string;
  content?: string;
}

export interface AuthHeaders {
  "x-ms-date": string;
  authorization: string;
}

export interface KeyResource {
  Token: string;
}

export interface IndexingPolicy {
  automatic: boolean;
  indexingMode: "consistent" | "lazy" | "none";
  includedPaths: any;
  excludedPaths: any;
  compositeIndexes?: any[];
  spatialIndexes?: any[];
  vectorIndexes?: VectorIndex[];
  fullTextIndexes?: FullTextIndex[];
}

export interface VectorIndex {
  path: string;
  type: "flat" | "diskANN" | "quantizedFlat";
  vectorIndexShardKey?: string[];
  indexingSearchListSize?: number;
  quantizationByteSize?: number;
}

export interface FullTextIndex {
  path: string;
}

export interface ComputedProperty {
  name: string;
  query: string;
}

export type ComputedProperties = ComputedProperty[];

export interface MaterializedView {
  id: string;
  _rid: string;
}

export interface MaterializedViewDefinition {
  definition: string;
  sourceCollectionId: string;
  sourceCollectionRid?: string;
}

export interface PartitionKey {
  paths: string[];
  kind: "Hash" | "Range" | "MultiHash";
  version: number;
  systemKey?: boolean;
}

export interface Statistic {
  documentCount: number;
  id: string;
  partitionKeys: any[];
  sizeInKB: number;
}

export interface QueryPreparationTimes {
  queryCompilationTime: any;
  logicalPlanBuildTime: any;
  physicalPlanBuildTime: any;
  queryOptimizationTime: any;
}

export interface RuntimeExecutionTimes {
  queryEngineExecutionTime: any;
  systemFunctionExecutionTime: any;
  userDefinedFunctionExecutionTime: any;
}

export interface QueryMetrics {
  clientSideMetrics: any;
  documentLoadTime: any;
  documentWriteTime: any;
  indexHitDocumentCount: number;
  indexLookupTime: any;
  outputDocumentCount: number;
  outputDocumentSize: number;
  queryPreparationTimes: QueryPreparationTimes;
  retrievedDocumentCount: number;
  retrievedDocumentSize: number;
  runtimeExecutionTimes: RuntimeExecutionTimes;
  totalQueryExecutionTime: any;
  vmExecutionTime: any;
}

export interface Offer {
  id: string;
  autoscaleMaxThroughput: number | undefined;
  manualThroughput: number | undefined;
  minimumThroughput: number | undefined;
  offerDefinition?: SDKOfferDefinition;
  offerReplacePending: boolean;
  instantMaximumThroughput?: number;
  softAllowedMaximumThroughput?: number;
  throughputBuckets?: ThroughputBucket[];
}

export interface ThroughputBucket {
  id: number;
  maxThroughputPercentage: number;
}

export interface SDKOfferDefinition extends Resource {
  offerVersion?: string;
  offerType?: string;
  content?: {
    offerThroughput: number;
    offerIsRUPerMinuteThroughputEnabled?: boolean;
    collectionThroughputInfo?: OfferThroughputInfo;
    offerAutopilotSettings?: AutoPilotOfferSettings;
  };
  resource?: string;
  offerResourceId?: string;
}

export interface OfferThroughputInfo {
  minimumRUForCollection: number;
  numPhysicalPartitions: number;
}

export interface UniqueKeyPolicy {
  uniqueKeys: UniqueKey[];
}

export interface UniqueKey {
  paths: string[];
}

export interface CreateDatabaseAndCollectionRequest {
  databaseId: string;
  collectionId: string;
  offerThroughput: number;
  databaseLevelThroughput: boolean;
  partitionKey?: PartitionKey;
  indexingPolicy?: IndexingPolicy;
  uniqueKeyPolicy?: UniqueKeyPolicy;
  autoPilot?: AutoPilotCreationSettings;
  analyticalStorageTtl?: number;
}

export interface AutoPilotCreationSettings {
  maxThroughput?: number;
}

export interface Query {
  id: string;
  resourceId: string;
  queryName: string;
  query: string;
}

export interface AutoPilotOfferSettings {
  maximumTierThroughput?: number;
  maxThroughput?: number;
  targetMaxThroughput?: number;
}

export interface CreateDatabaseParams {
  autoPilotMaxThroughput?: number;
  databaseId: string;
  databaseLevelThroughput?: boolean;
  offerThroughput?: number;
}

export interface CreateCollectionParamsBase {
  databaseId: string;
  databaseLevelThroughput: boolean;
  offerThroughput?: number;
  analyticalStorageTtl?: number;
  autoPilotMaxThroughput?: number;
  indexingPolicy?: IndexingPolicy;
  partitionKey?: PartitionKey;
  uniqueKeyPolicy?: UniqueKeyPolicy;
  createMongoWildcardIndex?: boolean;
  vectorEmbeddingPolicy?: VectorEmbeddingPolicy;
  fullTextPolicy?: FullTextPolicy;
}

export interface CreateCollectionParams extends CreateCollectionParamsBase {
  createNewDatabase: boolean;
  collectionId: string;
}

export interface CreateMaterializedViewsParams extends CreateCollectionParamsBase {
  materializedViewId: string;
  materializedViewDefinition: MaterializedViewDefinition;
}

export interface VectorEmbeddingPolicy {
  vectorEmbeddings: VectorEmbedding[];
}

export interface VectorEmbedding {
  dataType: "float16" | "float32" | "uint8" | "int8";
  dimensions: number;
  distanceFunction: "euclidean" | "cosine" | "dotproduct";
  path: string;
}

export interface FullTextPolicy {
  defaultLanguage: string;
  fullTextPaths: FullTextPath[];
}

export interface FullTextPath {
  path: string;
  language: string;
}

export interface ReadDatabaseOfferParams {
  databaseId: string;
  databaseResourceId?: string;
  offerId?: string;
}

export interface ReadCollectionOfferParams {
  collectionId: string;
  databaseId: string;
  collectionResourceId?: string;
  offerId?: string;
}

export interface UpdateOfferParams {
  currentOffer: Offer;
  databaseId: string;
  autopilotThroughput: number;
  manualThroughput: number;
  collectionId?: string;
  migrateToAutoPilot?: boolean;
  migrateToManual?: boolean;
  throughputBuckets?: ThroughputBucket[];
}

export interface Notification {
  id: string;
  kind: string;
  accountName: string;
  action: any;
  buttonValue?: any;
  collectionName?: string;
  databaseName?: string;
  description: string;
  endDateUtc: number;
  seenAtUtc: number;
  state: string;
  type: string;
  updatedAtUtc: string;
}

export enum ConflictResolutionMode {
  Custom = "Custom",
  LastWriterWins = "LastWriterWins",
}

/**
 * Represents the conflict resolution policy configuration for specifying how to resolve conflicts
 *  in case writes from different regions result in conflicts on documents in the collection in the Azure Cosmos DB service.
 */
export interface ConflictResolutionPolicy {
  /**
   * Gets or sets the ConflictResolutionMode in the Azure Cosmos DB service. By default it is ConflictResolutionMode.LastWriterWins.
   */
  mode?: keyof typeof ConflictResolutionMode;
  /**
   * Gets or sets the path which is present in each document in the Azure Cosmos DB service for last writer wins conflict-resolution.
   * This path must be present in each document and must be an integer value.
   * In case of a conflict occuring on a document, the document with the higher integer value in the specified path will be picked.
   * If the path is unspecified, by default the timestamp path will be used.
   *
   * This value should only be set when using ConflictResolutionMode.LastWriterWins.
   *
   * ```typescript
   * conflictResolutionPolicy.ConflictResolutionPath = "/name/first";
   * ```
   *
   */
  conflictResolutionPath?: string;
  /**
   * Gets or sets the StoredProcedure which is used for conflict resolution in the Azure Cosmos DB service.
   * This stored procedure may be created after the Container is created and can be changed as required.
   *
   * 1. This value should only be set when using ConflictResolutionMode.Custom.
   * 2. In case the stored procedure fails or throws an exception, the conflict resolution will default to registering conflicts in the conflicts feed.
   *
   * ```typescript
   * conflictResolutionPolicy.ConflictResolutionProcedure = "resolveConflict"
   * ```
   */
  conflictResolutionProcedure?: string;
}

export interface ChangeFeedPolicy {
  retentionDuration: number;
}

export interface GeospatialConfig {
  type: string;
}

export interface RegionEndpoint {
  name: string;
  documentAccountEndpoint: string;
}

export interface Tenant {
  displayName: string;
  id: string;
  tenantId: string;
  countryCode: string;
  domains: Array<string>;
}

export interface AccountKeys {
  primaryMasterKey: string;
  secondaryMasterKey: string;
  primaryReadonlyMasterKey: string;
  secondaryReadonlyMasterKey: string;
}

export interface OperationStatus {
  status: string;
  id?: string;
  name?: string; // operationId
  properties?: any;
  error?: { code: string; message: string };
}

export interface NotebookWorkspaceConnectionInfo {
  authToken: string;
  notebookServerEndpoint: string;
  forwardingId: string;
}

export interface ContainerInfo {
  durationLeftInMinutes: number;
  phoenixServerInfo: NotebookWorkspaceConnectionInfo;
  status: ContainerStatusType;
}

export interface IProvisionData {
  cosmosEndpoint?: string;
  poolId: string;
  databaseId?: string;
  containerId?: string;
  mode?: string;
}

export interface IContainerData {
  forwardingId: string;
}

export interface IDbAccountAllow {
  status: number;
  message?: string;
  type?: string;
}

export interface IResponse<T> {
  status: number;
  data: T;
}

export interface IPhoenixError {
  message: string;
  type: string;
}

export interface IMaxAllocationTimeExceeded extends IPhoenixError {
  earliestAllocationTimestamp: string;
  maxAllocationTimePerDayPerUserInMinutes: string;
}

export interface IMaxDbAccountsPerUserExceeded extends IPhoenixError {
  maxSimultaneousConnectionsPerUser: string;
}

export interface IMaxUsersPerDbAccountExceeded extends IPhoenixError {
  maxSimultaneousUsersPerDbAccount: string;
}

export interface IPhoenixConnectionInfoResult {
  readonly phoenixServiceInfo?: IPhoenixServiceInfo;
}

export interface IPhoenixServiceInfo {
  readonly authToken?: string;
  readonly phoenixServiceUrl?: string;
  readonly forwardingId?: string;
}

export interface NotebookWorkspaceFeedResponse {
  value: NotebookWorkspace[];
}

export interface NotebookWorkspace {
  id: string;
  name: string;
  properties: {
    status: string;
    notebookServerEndpoint: string;
  };
}

export interface NotebookConfigurationEndpoints {
  path: string;
  endpoints: NotebookConfigurationEndpointInfo[];
}

export interface NotebookConfigurationEndpointInfo {
  type: string;
  endpoint: string;
  username: string;
  password: string;
  token: string;
}

export interface SparkClusterConnectionInfo {
  userName: string;
  password: string;
  endpoints: SparkClusterEndpoint[];
}

export interface SparkClusterEndpoint {
  kind: SparkClusterEndpointKind;
  endpoint: string;
}

export enum SparkClusterEndpointKind {
  SparkUI = "SparkUI",
  HistoryServerUI = "HistoryServerUI",
  Livy = "Livy",
}

export interface RpParameters {
  db: string;
  offerThroughput?: number;
  st: Boolean;
  sid: string;
  rg: string;
  dba: string;
  partitionKeyVersion?: number;
}

export interface MongoParameters extends RpParameters {
  pk?: string;
  resourceUrl?: string;
  coll?: string;
  cd?: Boolean;
  is?: Boolean;
  rid?: string;
  rtype?: string;
  isAutoPilot?: Boolean;
  autoPilotThroughput?: string;
  analyticalStorageTtl?: number;
}

export interface MemoryUsageInfo {
  freeKB: number;
  totalKB: number;
}

export interface ContainerConnectionInfo {
  status: ConnectionStatusType;
  //need to add ram and rom info
}

export interface FirewallRule {
  id: string;
  name: string;
  type: string;
  properties: {
    startIpAddress: string;
    endIpAddress: string;
  };
}

export enum PhoenixErrorType {
  MaxAllocationTimeExceeded = "MaxAllocationTimeExceeded",
  MaxDbAccountsPerUserExceeded = "MaxDbAccountsPerUserExceeded",
  MaxUsersPerDbAccountExceeded = "MaxUsersPerDbAccountExceeded",
  AllocationValidationResult = "AllocationValidationResult",
  RegionNotServicable = "RegionNotServicable",
  SubscriptionNotAllowed = "SubscriptionNotAllowed",
  UnknownError = "UnknownError",
  PhoenixFlightFallback = "PhoenixFlightFallback",
  UserMissingPermissionsError = "UserMissingPermissionsError",
}

export interface CopilotEnabledConfiguration {
  isEnabled: boolean;
}

export interface FeatureRegistration {
  name: string;
  properties: {
    state: string;
  };
}

export type Tags = { [key: string]: string };
