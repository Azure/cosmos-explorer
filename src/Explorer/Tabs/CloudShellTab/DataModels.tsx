/**
 * Copyright (c) Microsoft Corporation.  All rights reserved.
 */

export const enum OsType {
    Linux = "linux",
    Windows = "windows"
}

export const enum ShellType {
    Bash = "bash",
    PowerShellCore = "pwsh"
}
  
export const enum NetworkType {
    Default = "Default",
    Isolated = "Isolated"
}

export const enum SessionType {
    Mounted = "Mounted",
    Ephemeral = "Ephemeral"
}

export type Settings = {
    location: string;
    sessionType: SessionType;
    osType: OsType;
};

export type ProvisionConsoleResponse = {
    properties: {
        osType: OsType;
        provisioningState: string;
        uri: string;
    };
};

export type Authorization = {
    token: string;
};

export type ConnectTerminalResponse = {
    id: string;
    idleTimeout: string;
    rootDirectory: string;
    socketUri: string;
    tokenUpdated: boolean;
};


