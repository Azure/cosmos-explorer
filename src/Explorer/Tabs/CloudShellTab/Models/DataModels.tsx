export const enum OsType {
  Linux = "linux",
  Windows = "windows",
}

export const enum ShellType {
  Bash = "bash",
  PowerShellCore = "pwsh",
}

export const enum NetworkType {
  Default = "Default",
  Isolated = "Isolated",
}

/**
 * Azure CloudShell session types:
 * - Mounted: Sessions with persistent storage via an Azure File Share mount.
 *   Files and configurations are preserved between sessions, allowing for
 *   continuity of work across multiple CloudShell sessions.
 *
 * - Ephemeral: Temporary sessions without persistent storage.
 *   All files and changes are discarded when the session ends.
 *   These sessions start faster but don't retain user data.
 *
 * The session type affects resource allocation, startup time,
 * and whether user files/configurations persist between sessions.
 */
export const enum SessionType {
  Mounted = "Mounted",
  Ephemeral = "Ephemeral",
}

export type CloudShellSettings = {
  properties: UserSettingProperties;
};

export type UserSettingProperties = {
  networkType: string;
  preferredLocation: string;
  preferredOsType: OsType;
  preferredShellType: ShellType;
  userSubscription: string;
  sessionType: SessionType;
  vnetSettings: object;
};

export type ProvisionConsoleResponse = {
  properties: {
    osType: OsType;
    provisioningState: string;
    uri: string;
  };
};

export type Authorization = {
  token: string;
};

export type ConnectTerminalResponse = {
  id: string;
  idleTimeout: string;
  rootDirectory: string;
  socketUri: string;
  tokenUpdated: boolean;
};

export type ProviderAuthorization = {
  applicationId: string;
  roleDefinitionId: string;
};

export type ProviderResourceType = {
  resourceType: string;
  locations: string[];
  apiVersions: string[];
  defaultApiVersion?: string;
  capabilities?: string;
};

export type RegistrationState = "Registered" | "NotRegistered" | "Registering" | "Unregistering";
export type RegistrationPolicy = "RegistrationRequired" | "RegistrationOptional";

export type CloudShellProviderInfo = {
  id: string;
  namespace: string;
  authorizations?: ProviderAuthorization[];
  resourceTypes: ProviderResourceType[];
  registrationState: RegistrationState;
  registrationPolicy: RegistrationPolicy;
};
