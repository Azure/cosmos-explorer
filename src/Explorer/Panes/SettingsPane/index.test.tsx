import { shallow } from "enzyme";
import React from "react";
import { SettingsPane } from ".";
import { DatabaseAccount } from "../../../Contracts/DataModels";
import { updateUserContext } from "../../../UserContext";
const props = {
  closePanel: (): void => undefined,
  openNotificationConsole: (): void => undefined,
};
describe("Settings Pane", () => {
  it("should render Default properly", () => {
    const wrapper = shallow(<SettingsPane {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Gremlin properly", () => {
    updateUserContext({
      databaseAccount: {
        properties: {
          capabilities: [{ name: "EnableGremlin" }],
        },
      } as DatabaseAccount,
    });
    const wrapper = shallow(<SettingsPane {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render an setting as heading ", () => {
    const wrapper = shallow(<SettingsPane {...props} />);
    expect(wrapper.find('span[role="heading"]').text()).toEqual("Settings");
  });
});
