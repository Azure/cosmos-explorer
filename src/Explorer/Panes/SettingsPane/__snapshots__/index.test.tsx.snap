// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Pane should render Default properly 1`] = `
<div>
  <div
    className="contextual-pane-out"
  />
  <div
    className="contextual-pane"
    id="settingspane"
  >
    <div
      className="contextual-pane-in"
    >
      <form
        className="paneContentContainer"
      >
        <div
          className="firstdivbg headerline"
        >
          <span
            aria-level={2}
            role="heading"
          >
            Settings
          </span>
          <div
            aria-label="Close pane"
            className="closeImg"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            <img
              alt="Close"
              src=""
              title="Close"
            />
          </div>
        </div>
        <div
          className="paneMainContent"
        >
          <div>
            <div
              className="settingsSection"
            >
              <div
                className="settingsSectionPart"
              >
                <div
                  className="settingsSectionLabel"
                >
                  Enable cross-partition query
                  <Tooltip>
                    Send more than one request while executing a query. More than one request is necessary if the query is not scoped to single partition key value.
                  </Tooltip>
                </div>
                <StyledCheckboxBase
                  ariaLabel="Enable cross partition query"
                  checked={true}
                  className="padding"
                  onChange={[Function]}
                  style={
                    Object {
                      "padding": "0",
                    }
                  }
                  tabIndex={0}
                />
              </div>
            </div>
            <div
              className="settingsSection"
            >
              <div
                className="settingsSectionPart"
              >
                <div
                  className="settingsSectionLabel"
                >
                  Max degree of parallelism
                  <Tooltip>
                    Gets or sets the number of concurrent operations run client side during parallel query execution. A positive property value limits the number of concurrent operations to the set value. If it is set to less than 0, the system automatically decides the number of concurrent operations to run.
                  </Tooltip>
                </div>
                <CustomizedSpinButton
                  ariaLabel="Max degree of parallelism"
                  className="textfontclr"
                  decrementButtonIcon={
                    Object {
                      "iconName": "ChevronDownSmall",
                    }
                  }
                  disabled={false}
                  id="max-degree"
                  incrementButtonIcon={
                    Object {
                      "iconName": "ChevronUpSmall",
                    }
                  }
                  label=""
                  labelPosition={2}
                  max={100}
                  min={-1}
                  onChange={[Function]}
                  role="textbox"
                  step={1}
                  tabIndex={0}
                  value="6"
                />
              </div>
            </div>
            <div
              className="settingsSection"
            >
              <div
                className="settingsSectionPart"
              >
                <div
                  className="settingsSectionLabel"
                >
                  Explorer Version
                </div>
                <div />
              </div>
            </div>
          </div>
        </div>
        <div
          className="paneFooter"
        >
          <div
            className="leftpanel-okbut"
          >
            <CustomizedPrimaryButton
              onClick={[Function]}
              type="submit"
            >
              Apply
            </CustomizedPrimaryButton>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Settings Pane should render Gremlin properly 1`] = `
<div>
  <div
    className="contextual-pane-out"
  />
  <div
    className="contextual-pane"
    id="settingspane"
  >
    <div
      className="contextual-pane-in"
    >
      <form
        className="paneContentContainer"
      >
        <div
          className="firstdivbg headerline"
        >
          <span
            aria-level={2}
            role="heading"
          >
            Settings
          </span>
          <div
            aria-label="Close pane"
            className="closeImg"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            <img
              alt="Close"
              src=""
              title="Close"
            />
          </div>
        </div>
        <div
          className="paneMainContent"
        >
          <div>
            <div
              className="settingsSection"
            >
              <div
                className="settingsSectionPart"
              >
                <div
                  className="settingsSectionLabel"
                >
                  Display Gremlin query results as:Â 
                  <Tooltip>
                    Select Graph to automatically visualize the query results as a Graph or JSON to display the results as JSON.
                  </Tooltip>
                </div>
                <div
                  aria-label="Graph Auto-visualization"
                  className="tabs"
                  role="radiogroup"
                >
                  <div
                    className="tab"
                  >
                    <input
                      autoFocus={true}
                      checked={true}
                      defaultValue="false"
                      id="graphAutoVizOn"
                      name="graphAutoVizOption"
                      onChange={[Function]}
                      type="radio"
                    />
                    <label
                      aria-checked={true}
                      htmlFor="graphAutoVizOn"
                      id="graph-display"
                      onKeyPress={[Function]}
                      role="radio"
                      tabIndex={0}
                    >
                      Graph
                    </label>
                  </div>
                  <div
                    className="tab"
                  >
                    <input
                      checked={true}
                      defaultValue="true"
                      id="graphAutoVizOff"
                      name="graphAutoVizOption"
                      onChange={[Function]}
                      type="radio"
                    />
                    <label
                      aria-checked={false}
                      htmlFor="graphAutoVizOff"
                      onKeyPress={[Function]}
                      role="radio"
                      tabIndex={0}
                    >
                      JSON
                    </label>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="settingsSection"
            >
              <div
                className="settingsSectionPart"
              >
                <div
                  className="settingsSectionLabel"
                >
                  Explorer Version
                </div>
                <div />
              </div>
            </div>
          </div>
        </div>
        <div
          className="paneFooter"
        >
          <div
            className="leftpanel-okbut"
          >
            <CustomizedPrimaryButton
              onClick={[Function]}
              type="submit"
            >
              Apply
            </CustomizedPrimaryButton>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
